src/Data/HashMap/Linear.hs

```diff
@@ -224,14 +226,15 @@ member hmap k = memberFromQuery (queryIndex hmap k)
     memberFromQuery (h, IndexToSwap _ _) = (h, False)
     memberFromQuery (h, IndexToUpdate _ _) = (h, True)
 
-lookup :: Keyed k => HashMap k v #-> k -> (HashMap k v, Maybe v)
+lookup :: Keyed k => HashMap k v #-> k -> (HashMap k v, Unrestricted (Maybe v))
```

Comment:
Asking the tough questions: do we want `Unrestricted (Maybe v)` or `Maybe (Unrestriced v)`, or, even, some maybe variant which has an unrestricted field?

Reply:
This is a tough question ... hmm

Reply:
I'll make this an issue and defer the problem.